pkg/gitops/flux/installer.go

```diff
@@ -117,27 +116,29 @@ func (fi *Installer) Run(ctx context.Context) (string, error) {
 	logger.Info("Waiting for Flux to start")
 	err = waitForFluxToStart(fi.opts.Operator.Namespace, fi.timeout, fi.k8sClientSet)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 	logger.Info("fetching public SSH key from Flux")
 	fluxSSHKey, err := getPublicKeyFromFlux(ctx, fi.opts.Operator.Namespace, fi.timeout, fi.k8sRestConfig, fi.k8sClientSet)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	logger.Info("Flux started successfully")
 	logger.Info("see https://docs.fluxcd.io/projects/flux for details on how to use Flux")
 
-	logger.Info("Committing and pushing manifests to %s", fi.opts.Repo.URL)
-	if err = fi.addFilesToRepo(); err != nil {
-		return "", err
+	if !fi.opts.ReadOnly {
```

Comment:
Why is this conditioned by readonly?

Reply:
I thought the intention of the user would be that they won't like eksctl to write to the repo as well if you want Flux to be read-only. Is that a right assumption?

If not, maybe we'd better split this to two configuration fields, the one for eksctl as e.g. `git.commitManifestsOnStart` and one for flux as e.g.`git.operator.readOnly`.

WDYT?

Reply:
Anyway, I've added an additional commit to implement what I think best now based on your review: 18f7d768

It's now:

```yaml

git:
  # NEW: Commit and push Flux manifests to the Git Repo on install
  # The default value is true for backward-compatbiility, so explicitly set to `false `to disable
  #commitOperatorManifests: false
  repo:
    url: "git@github.com:mumoshu/gitops-demo.git"
    branch: master
    fluxPath: "flux/"
    user: "gitops"
    email: "gitops@myorg.com"
    privateSSHKeyPath: /Users/c-ykuoka/.ssh/id_rsa
  operator:
    namespace: "flux"
    # NEW: Instruct Flux to read-only mode and create the deploy key as read-only
    readOnly: true
```

And commands like `eksctl enable repo` has been given respective common flux flags:

```
      --read-only                         Instruct Flux to read-only mode and create the deploy key as read-only
      --commit-operator-manifests         Commit and push Flux manifests to the Git Repo on install (default true)
```